packages/ember-data/lib/serializers/rest-serializer.js

```diff
@@ -683,12 +714,72 @@ var RESTSerializer = JSONSerializer.extend({
   serializePolymorphicType: function(snapshot, json, relationship) {
     var key = relationship.key;
     var belongsTo = snapshot.belongsTo(key);
+    var typeKey = this.keyForPolymorphicType(key, relationship.type, 'serialize');
+
+    // old way of getting the key for the polymorphic type
     key = this.keyForAttribute ? this.keyForAttribute(key, "serialize") : key;
+    key = `${key}Type`;
+
+    // check if the old way differs with the new, correct one, and if so, log a
+    // deprecation warning and restore the old behavior
+    if (key !== typeKey) {
+      Ember.deprecate("The typeKey to serialize a polymorphic type uses keyForAttribute which has been deprecated in favor of keyForPolymorphicType.", {
+        id: 'ds.rest-serializer.deprecated-key-for-polymorphic-type',
+        until: '3.0.0'
+      });
+
+      typeKey = key;
+    }
```

Comment:
Actually, I think this would be save without the deprecation, since there wouldn't have been another way to change the way the polymorphic record is serialized except for overriding `serializePolymorphicType` as a whole...